### PR TITLE
Add coroutines/flow support

### PR DESCRIPTION
**Pull request by Jawnnypoo**
This adds support for Coroutines/Flow, mostly matching the offerings with the rxjava3 module, with a few parallels missing which are called out in the README of the module.

### CODE REVIEW

**Comment by Jawnnypoo** [comment]
Be warned that upgrading from Gradle 7.5 to the latest (8.8) results in errors in the jacoco plugin, so we would have to resolve that if we wanted to update Gradle.

File: build.gradle [hunk 1]
```diff
@@ -59,8 +46,7 @@ task clean(type: Delete) {
 
 apply plugin: "com.vanniktech.android.junit.jacoco"
 junitJacoco {
-    // WARNING! Don't upgrade unless https://github.com/Malinskiy/danger-jacoco/issues/10 is fixed
-    jacocoVersion = '0.8.2'
+    jacocoVersion = '0.8.12'
```
Comment (Jawnnypoo) [inline build.gradle]: Saw issues without upgrading this.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,7 @@
 # Changelog
+## [0.4.0] - June 3, 2024
+- Added `lce-flow` module for Coroutine Flow support.
```
Comment (Laimiux) [inline CHANGELOG.md]: Nit: `lce-coroutines`

File: lce-coroutines/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# lce-coroutines
+Coroutines/Flow support for LCE. 
+
+If coming over from RxJava, here is the typical mapping
+
+- `switchMapContent` -> `flatMapLatestContent`
```
Comment (Laimiux) [inline lce-coroutines/README.md:6]: Does `flatMapLatestContent` cancel previously running flow?

Comment (Jawnnypoo) [reply lce-coroutines/README.md:6]: Yes, thats correct - https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/flat-map-latest.html

File: CHANGELOG.md (hunk 2 above)
Comment (Jawnnypoo) [reply CHANGELOG.md]: Good catch, I was back and forth on the name of this. Fixed.

File: .github/workflows/checks.yml [hunk 4]
```diff
@@ -12,9 +12,10 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - uses: gradle/wrapper-validation-action@v1
-      - uses: actions/setup-java@v1
+      - uses: actions/setup-java@v3
         with:
-          java-version: 11
+          distribution: 'zulu'
+          java-version: 17
```
Comment (Laimiux) [inline .github/workflows/checks.yml:18]: Can you move java and library updates to a separate PR?

Comment (Jawnnypoo) [reply .github/workflows/checks.yml:18]: Yep!

Comment (Jawnnypoo) [reply .github/workflows/checks.yml:18]: https://github.com/Laimiux/lce/pull/44

File: README.md [hunk 5]
```diff
@@ -112,6 +112,15 @@ fun fetchData(): Observable<UCT<MyData>> {
 ```
 You'll need to add the `lce-rxjava3` dependency to access these features.
 
+### Coroutines Support
+Converting a `Flow<T>` operation into `Flow<UCT<T>>`
+```kotlin
+fun fetchData(): Flow<UCT<MyData>> = flow {
+    emit(MyData()).toUCT()
```
Comment (Laimiux) [inline README.md]: Does `emit` returns a `Flow`?

File: lce-coroutines/src/main/java/com/laimiux/lce/flow/Init.kt [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import com.laimiux.lce.LC
+import com.laimiux.lce.UC
+import com.laimiux.lce.UCE
+import com.laimiux.lce.UCT
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onStart
+
+inline fun <C : Any, E> Flow<C>.toUCE(
```
Comment (Laimiux) [inline lce-coroutines/src/main/java/com/laimiux/lce/flow/Init.kt]: Does `Flow` support nullable events?

File: gradle.properties [hunk 7]
```diff
@@ -1,5 +1,5 @@
 GROUP=com.laimiux.lce
-VERSION_NAME=0.3.4
+VERSION_NAME=0.4.0
```
Comment (Laimiux) [inline gradle.properties]: I don't think we should update version name in this pull-request

File: README.md (hunk 5 above)
Comment (Jawnnypoo) [reply README.md]: Not exactly, the `= flow {` is the part that makes it return a Flow

File: lce-coroutines/src/main/java/com/laimiux/lce/flow/Init.kt (hunk 6 above)
Comment (Jawnnypoo) [reply lce-coroutines/src/main/java/com/laimiux/lce/flow/Init.kt]: Good catch, yep, will update this to allow for null

File: lce-coroutines/gradle.properties [hunk 8]
```diff
@@ -0,0 +1,3 @@
+POM_ARTIFACT_ID=lce-coroutines
+POM_NAME=LCE type Flow Support
```
Comment (Laimiux) [inline lce-coroutines/gradle.properties]: nit: `LCE Kotlin Coroutines Support`

File: README.md (hunk 5 above)
Comment (Laimiux) [reply README.md]: Is this documentation right? Could you add a couple of unit tests?

File: lce-rxjava3/src/main/java/com/laimiux/lce/rxjava3/Init.kt [hunk 9]
```diff
@@ -53,7 +53,7 @@ fun <C : Any> Maybe<C>.toUCT(): Observable<UCT<C>> {
 
 fun <C : Any> Observable<C>.toUCT(): Observable<UCT<C>> {
     return this
-        .map { UCT.content(it) as UCT<C> }
+        .map { UCT.content(it) }
```
Comment (Laimiux) [inline lce-rxjava3/src/main/java/com/laimiux/lce/rxjava3/Init.kt]: Nit: Not exactly related

File: README.md (hunk 5 above)
Comment (Jawnnypoo) [reply README.md]: Done. It was a good idea, as there were some issues with the emission of loading and error state values.

**Review by Laimiux — APPROVED** [review]
LGTM!
